File: ZooJava/src/main/java/com/example/zoojava/models/Animal.java
package com.example.zoojava.models;

import com.example.zoojava.models.enums.typeAnimal;
import javafx.beans.property.*;

import java.time.LocalDate;

/**
 * Clase de creación de los animales del zoo.
 */
public class Animal {
    private IntegerProperty id;
    private StringProperty name;
    private ObjectProperty<typeAnimal> type;
    private ObjectProperty<LocalDate> birthDate;
    private StringProperty img;


    public Animal(int id, String name, typeAnimal type, LocalDate birthDate, String img) {
        this.id = new SimpleIntegerProperty(id);
        this.name = new SimpleStringProperty(name);
        this.type = new SimpleObjectProperty<>(type);
        this.birthDate = new SimpleObjectProperty<>(birthDate);
        this.img = new SimpleStringProperty(img);
    }

    public void setId(int id) {
        this.id.set(id);
    }

    public int getId() {
        return id.get();
    }

    public IntegerProperty idProperty() {
        return id;
    }


    public String getName() {
        return name.get();
    }

    public StringProperty nameProperty() {
        return name;
    }

    public void setName(String name) {
        this.name.set(name);
    }

    public typeAnimal getType() {
        return type.get();
    }

    public ObjectProperty<typeAnimal> typeProperty() {
        return type;
    }

    public void setType(typeAnimal type) {
        this.type.set(type);
    }

    public LocalDate getBirthDate() {
        return birthDate.get();
    }

    public ObjectProperty<LocalDate> birthDateProperty() {
        return birthDate;
    }

    public void setBirthDate(LocalDate birthDate) {
        this.birthDate.set(birthDate);
    }

    public String getImg() {
        return img.get();
    }

    public StringProperty imgProperty() {
        return img;
    }

    public void setImg(String img) {
        this.img.set(img);
    }

    public String toCsv() {
        return getId()+";"+getName()+";"+getType().toString()+";"+getBirthDate().toString()+";"+getImg();
    }
}

File: ZooKotlin/src/main/kotlin/com/example/zookotlin/models/Animal.kt
package com.example.zookotlin.models

import com.example.zookotlin.models.enums.typeAnimal
import javafx.beans.property.*
import java.time.LocalDate

/**
 * Clase creacion modelo aniamles.
 */
class Animal {
    private var id: IntegerProperty
    private var name: StringProperty
    private var type: ObjectProperty<typeAnimal>
    private var birthDate: ObjectProperty<LocalDate>
    private var img: StringProperty


    constructor(id: Int, name: String, type: typeAnimal, birthDate: LocalDate, img: String){
        this.id = SimpleIntegerProperty(id)
        this.name = SimpleStringProperty(name)
        this.type = SimpleObjectProperty(type)
        this.birthDate = SimpleObjectProperty(birthDate)
        this.img = SimpleStringProperty(img)
    }


    fun setId(id: Int) {
        this.id.set(id)
    }

    fun getId(): Int {
        return id.get()
    }

    fun idProperty(): IntegerProperty {
        return id
    }


    fun getName(): String {
        return name.get()
    }

    fun nameProperty(): StringProperty {
        return name
    }

    fun setName(name: String) {
        this.name.set(name)
    }

    fun getType(): typeAnimal {
        return type.get()
    }

    fun typeProperty(): ObjectProperty<typeAnimal> {
        return type
    }

    fun setType(type: typeAnimal?) {
        this.type.set(type)
    }

    fun getBirthDate(): LocalDate {
        return birthDate.get()
    }

    fun birthDateProperty(): ObjectProperty<LocalDate> {
        return birthDate
    }

    fun setBirthDate(birthDate: LocalDate?) {
        this.birthDate.set(birthDate)
    }

    fun getImg(): String {
        return img.get()
    }

    fun imgProperty(): StringProperty {
        return img
    }

    fun setImg(img: String) {
        this.img.set(img)
    }


    /**
     * Para pasar a csv todos los animales.
     */
    fun toCsv(): String {
        return getId().toString() + ";" + getName() + ";" + getType().toString() + ";" + getBirthDate().toString() + ";" + getImg()
    }
}